data-protocols/dsp/dsp-catalog/dsp-catalog-http-api/src/main/java/org/eclipse/edc/protocol/dsp/catalog/http/api/decorator/CatalogPaginationResponseDecorator.java

```diff
@@ -44,19 +44,19 @@ public CatalogPaginationResponseDecorator(String requestUrl, ContinuationTokenSe
     public Response.ResponseBuilder decorate(Response.ResponseBuilder responseBuilder, CatalogRequestMessage requestBody, Catalog responseBody) {
         var currentQuerySpec = requestBody.getQuerySpec();
         if (responseBody.getDatasets().size() == currentQuerySpec.getLimit()) {
-            addLink(NEXT, responseBuilder, currentQuerySpec, (offset, limit) -> offset + limit);
+            addLink(NEXT, requestBody.getProtocol(), responseBuilder, currentQuerySpec, (offset, limit) -> offset + limit);
         }
 
         if (currentQuerySpec.getOffset() >= currentQuerySpec.getLimit()) {
-            addLink(PREV, responseBuilder, currentQuerySpec, (offset, limit) -> offset - limit);
+            addLink(PREV, requestBody.getProtocol(), responseBuilder, currentQuerySpec, (offset, limit) -> offset - limit);
         }
 
         return responseBuilder;
     }
 
-    private void addLink(String rel, Response.ResponseBuilder responseBuilder, QuerySpec currentQuerySpec, IntBinaryOperator newOffsetOperator) {
+    private void addLink(String rel, String protocol, Response.ResponseBuilder responseBuilder, QuerySpec currentQuerySpec, IntBinaryOperator newOffsetOperator) {
         var newOffset = newOffsetOperator.applyAsInt(currentQuerySpec.getOffset(), currentQuerySpec.getLimit());
-        continuationTokenSerDes.serialize(currentQuerySpec.toBuilder().offset(newOffset).build())
+        continuationTokenSerDes.serialize(currentQuerySpec.toBuilder().offset(newOffset).build(), protocol)
```

Comment:
could the `continuationTokenSerDes` instance already contain the correct transformer registry as field without the need to break the interface? (by instantiate two different instances that will be passed to the two different controllers)